icu4c/source/i18n/currunit.cpp

```diff
@@ -50,6 +52,30 @@ CurrencyUnit::CurrencyUnit(ConstChar16Ptr _isoCode, UErrorCode& ec) {
     initCurrency(simpleIsoCode);
 }
 
+CurrencyUnit::CurrencyUnit(StringPiece _isoCode, UErrorCode& ec) {
+    // Note: unlike the old constructor, reject empty arguments with an error.
+    char isoCodeBuffer[4];
+    const char* isoCodeToUse;
+    // uprv_memchr checks that the string contains no internal NULs
+    if (_isoCode.length() != 3 || uprv_memchr(_isoCode.data(), 0, 3) != nullptr) {
```

Comment:
I don't think we have a `uprv_memchr` though. Perhaps we should add one?
(Assuming you mean a wrapper over [memchr](https://en.cppreference.com/w/c/string/byte/memchr)?)

Reply:
Yeah, I added it but forgot to commit the file.  I just copied how we implement uprv_memcpy:

```cpp
#define uprv_memset(buffer, mark, size) U_STANDARD_CPP_NAMESPACE memset(buffer, mark, size)
#define uprv_memcmp(buffer1, buffer2, size) U_STANDARD_CPP_NAMESPACE memcmp(buffer1, buffer2,size)
#define uprv_memchr(ptr, value, num) U_STANDARD_CPP_NAMESPACE memchr(ptr, value, num)
```

Commit pushed.